integration-libs/opf/base/core/facade/opf-payment.service.ts

```diff
@@ -45,4 +186,56 @@ export class OpfPaymentService implements OpfPaymentFacade {
       paymentVerificationPayload,
     });
   }
+
+  submitPayment(submitInput: SubmitInput): Observable<boolean> {
+    return this.submitPaymentCommand.execute({ submitInput });
+  }
+
+  protected displayError(error: PaymentError | undefined): void {
+    this.globalMessageService.add(
+      {
+        key: error?.message ? error.message : this.defaultError.message,
+      },
+      GlobalMessageType.MSG_TYPE_ERROR
+    );
+  }
+
+  protected handleBadRequestError(errorType: string): string {
```

Comment:
There are several methods related to error handling in this file. Perhaps it would be worthwhile to create a service that consolidates all these methods, allowing us to reuse them in various contexts. However, I'm unsure if we should implement this in the current pull request, as it might necessitate a broader perspective on how to address it across the entire OPF

Reply:
creating a opf-payment-error-handler.service.ts in opf/Base core/services folder, that will be use for the 2 other sub-patterns. I think no need to be in root as it is used within opf/Base facades